curp/src/rpc/connect.rs

```diff
@@ -127,9 +154,10 @@ impl ConnectInterface for Connect {
         request: ProposeRequest,
         timeout: Duration,
     ) -> Result<tonic::Response<ProposeResponse>, ProposeError> {
-        #[cfg(test)]
-        if !self.reachable.load(Ordering::Relaxed) {
-            return Err(ProposeError::RpcError("unreachable".to_owned()));
+        if let Some(ref f) = self.tx_filter {
```

Comment:
We can pack this code as a method `check_filter` or something like that. And call it here to reduce some duplicated code

Reply:
Done.